sources/interface/BankDialog.cpp

```diff
@@ -0,0 +1,143 @@
+#include "BankDialog.h"
+#include "ui_BankDialog.h"
+#include <iostream>
+#include "stdio.h"
+
+BankDialog::BankDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::BankDialog)
+{
+    ui->setupUi(this);
+    QStringList labelsTree = {"Num", "Name"};
+    ui->treeWidget->setHeaderLabels(labelsTree);
+}
+
+BankDialog::~BankDialog()
+{
+    delete ui;
+}
+
+void BankDialog::init(Scene *currentScene, int l_selectMode, int l_midiChannel)
+{
+    bankScene = currentScene;
+    resetCombiNames();
+    selectMode = l_selectMode;
+    selectedMidiChannel = l_midiChannel;
+}
+
+void BankDialog::resetCombiNames()
+{
+    ui->treeWidget->clear();
+    ui->treeWidget->setSortingEnabled(false);
+    for (int i = 0; i <= 127; i++){
+        QTreeWidgetItem *itm = new QTreeWidgetItem(ui->treeWidget);
+        if (bankScene->m_midiBank.existCombi(i))
+            itm->setText(1, bankScene->m_midiBank.findCombi(i).getName());
+        else
+            itm->setText(1, "<..>");
+        QString l_num = QString::number(i);
+        for (unsigned j = 0; j = (3 - l_num.length()); j++){
+            l_num = "0" + l_num;
+        }
+        itm->setText(0, l_num);
+        ui->treeWidget->addTopLevelItem(itm);
+    }
+    ui->treeWidget->setSortingEnabled(true);
+    ui->treeWidget->sortItems(0, Qt::AscendingOrder);
+}
+
+void BankDialog::on_treeWidget_itemClicked(QTreeWidgetItem *item, int column)
+{
+    selectedCombi = item->text(0).toInt();
+}
+
+void BankDialog::on_treeWidget_itemDoubleClicked(QTreeWidgetItem *item, int column)
+{
+    switch (selectMode){
+    case EDIT: {
+        selectedCombi = item->text(0).toInt();
+        if (selectedCombi >= 0){
+            if (! bankScene->m_midiBank.existCombi(selectedCombi))
+                bankScene->m_midiBank.createCombi(selectedCombi);
```

Comment:
Cette opération devrait être réalisée sous la possession du mutex de scène.
Le conteneur `QMap` n'est pas automatiquement thread-safe ; il existe donc une possibilité de corruption lorsque le fil audio lit la map alors que l'interface écrit.

Reply:
L'exemple se trouve ici.
https://github.com/linuxmao-org/Frontieres/blob/26a67f57fd17904a9e14916c9b1704188f520101/sources/interface/MyGLApplication.cpp#L113-L116

Reply:
ok, je peux entourer donc l'appel par ces deux lignes, mais ça ne me suffit pas, ça comme info pour etre capable de le faire : 
non seulement je ne sais pas quels sont les include necessaires (encore que avec un include <mutex> j'aie pu le faire, mais ensuite j'ai un message '::currentSceneMutex' has not been declared
et ça non plus je ne sais pas ou on en trouve la declaration, ou s'il faut la faire, comment.

Reply:
`currentSceneMutex` est défini dans `Frontieres.h`